Constitution parser

JacobKeegan opened the pull request:
Parser for the Constitution.
Parser for the Senate and Assembly Rules.
Fixes all empty titles. May still be a few mistakes (can't realistically check them all).

JacobKeegan committed:
Some initial changes. Attempting to isolate special chapter differences.

JacobKeegan committed:
First attempt at a ConstitutionBuilder.

JacobKeegan committed:
Merge remote-tracking branch 'origin/dev' into dev

JacobKeegan committed:
Changed LawNodeView to go better with Constitution structure.
Changed AbstractLawBuilder to adjust for Constitution structure.
Continued developing ConstitutionBuilder.

JacobKeegan committed:
Now parses section titles correctly, sans sections of the form #-(letter).

JacobKeegan committed:
That last commit was a lie, at the time, but now it's fixed.

JacobKeegan committed:
Now parses earlier specified sections correctly, and sections 36 onwards of Article 6.
Remaining bug at Article 6 section 23 due to random numbers.

JacobKeegan committed:
Commit before push.

JacobKeegan committed:
Everything now processes correctly, sans when a new title is gven after the document is already received.

JacobKeegan committed:
Some cleanup.

JacobKeegan committed:
Fixed last bugs, it seems to be finished.

JacobKeegan committed:
First attempt at a parser for rules.

JacobKeegan committed:
Bug fix for the ConstitutionParser, ensuring that documents are always processed in the same order.

JacobKeegan committed:
RulesBuilder is now functioning, except for Joint Rules (not sure what to do with them yet).

JacobKeegan committed:
Fixed bug #12793. After checking everything, only the preamble needed to change how it's parsed.
Some miscellaneous clean up too.

JacobKeegan committed:
Checking for more bugs. Working on lowering number of section title mismatches.

JacobKeegan committed:
Merge branches 'ConstitutionParser' and 'dev' of https://github.com/nysenate/OpenLegislation into ConstitutionParser

JacobKeegan committed:
Removed testing for, and implemented bug fix of, bug #12793.

JacobKeegan committed:
LawTitleParser passes all tests.

JacobKeegan committed:
EPT section titles now get parsed correctly.

JacobKeegan committed:
Greatly reduced mismatches.

JacobKeegan committed:
Fixed some Constitution and rules mismatches.
Added String to display if no title is given (vs. when the title couldn't be figured out).

JacobKeegan committed:
Simplified and upgraded section parsing. Down to 30 mismatches from the original DATABASE files.

JacobKeegan committed:
Suppressed warnings about GCMCUBIT.
Manually handled bad data in the Constitution.
Fixed GBSA29-AAA mistakenly being labeled as the child of another article.
Handled some more non-section title parsing cases.
Down to only 2 mismatches.

JacobKeegan committed:
Cleaned up some code. Added a lot of stuff to properly handle the GCT sections. All titles with asterisks in them are now handled properly.

JacobKeegan committed:
Improved the parser. Accounted for some discrepancies in formatting. Cleaned up some code. Now down to 6 mismatches across all the updates.

JacobKeegan committed:
Should've fixed the last title mismatch, a Constitution problem. Added in support for Joint Rules in RulesBuilder. Added in a labelID to always display numbers (seems to not be working, just need it in the right place I think).

KevinCaseiras commented:
This looks good.
Lets add back the `.idea/codeStyleSettings.xml` file though. It configures some default code styles shared by all developers on this project.

JacobKeegan commented:
I don't remember why I removed that file, so sounds good to me!

JacobKeegan committed:
Added back a file.

JacobKeegan committed:
Removed labelId. LawTitleParser now gets the docTypeId in the text directly.

JacobKeegan committed:
Just making the spacing the same as before.

JacobKeegan committed:
Just making the spacing the same as before.

JacobKeegan committed:
A couple more bad data instances.

KevinCaseiras committed:
Merge branch 'dev' into ConstitutionParser

KevinCaseiras committed:
Fix a few errors with law processing

KevinCaseiras committed:
Improve performance of data process api

refs #12315

KevinCaseiras committed:
Revert changes to package.json

KevinCaseiras committed:
Remove hard coded law titles

Hard codeded law titles could cause problems in the future if the titles
ever change. For now, I have documented these edge cases and we can
discuss other options for handling them.
